osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs

```diff
@@ -108,19 +108,13 @@ public void TestReadToEndNoPeeks()
         [Test]
         public void TestReadToEndAfterReadsAndPeeks()
         {
-            const string contents = "this line is gone\rthis one shouldn't be\r\nthese ones\ndefinitely not";
+            const string contents = "first line\r\nsecond line";
 
             using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(contents)))
             using (var bufferedReader = new LineBufferedReader(stream))
             {
-                Assert.AreEqual("this line is gone", bufferedReader.ReadLine());
-                Assert.AreEqual("this one shouldn't be", bufferedReader.PeekLine());
-
-                string[] endingLines = bufferedReader.ReadToEnd().Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
-                Assert.AreEqual(3, endingLines.Length);
-                Assert.AreEqual("this one shouldn't be", endingLines[0]);
-                Assert.AreEqual("these ones", endingLines[1]);
-                Assert.AreEqual("definitely not", endingLines[2]);
+                bufferedReader.PeekLine();
+                Assert.Throws<InvalidOperationException>(() => bufferedReader.ReadToEnd());
```

Comment:
I think a better way to handle that case would be to just reset the position of the stream in `ReadToEnd`, ie. rename it to `ReadAllContent`. Would that work for you?

Reply:
Only if that's also what `StreamReader` does. Speaking of, what is this class actually even doing anymore? It's just... wrapping a `StreamReader` but otherwise doing nothing special?

Reply:
It allows peeking at lines, that's its only purpose.

> Only if that's also what `StreamReader` does.

Can you explain what you mean by that? I would be resetting the position of the `StreamReader` if that's what you are asking for.

Reply:
Well `StreamReader` doesn't reset position to 0. It's [documented](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoend?view=net-6.0) as "Reads all characters from the current position to the end of the stream", and I'd expect the same behaviour here. Not from "from the start of the file".

So in other words I'd expect it to return the full peeked line + any content following it, and no content prior to the peeked line.

Reply:
Fair, I've reverted the behaviour.